src/main/scala/com/ing/wbaa/airlock/proxy/api/PostRequestActions.scala

```diff
@@ -34,18 +40,39 @@ trait PostRequestActions {
       case _ => Future.successful(Done)
     }
 
-  private[this] def updateBucketPolicy(httpRequest: HttpRequest, s3Request: S3Request): Unit = {
+  private[this] def updateBucketPermissions(httpRequest: HttpRequest, s3Request: S3Request): Future[Done] = {
     lazy val bucketName = bucketRegex findFirstMatchIn httpRequest.uri.path.toString() map (_.group(1))
     if (httpRequest.method == HttpMethods.PUT &&
       bucketName.isDefined) {
-      setDefaultBucketPolicy(bucketName.get)
+      setDefaultBucketAcl(bucketName.get) map (_ => Done)
+    } else {
+      Future.successful(Done)
     }
   }
 
-  protected[this] def handlePostRequestActions(response: HttpResponse, httpRequest: HttpRequest, s3Request: S3Request, userSTS: User): Unit = {
-    createAtlasLineage(response, httpRequest, userSTS, s3Request.clientIPAddress)
-    createBucketNotification(response, httpRequest, s3Request, userSTS)
-    updateBucketPolicy(httpRequest, s3Request)
+  protected[this] def handlePostRequestActions(response: HttpResponse, httpRequest: HttpRequest, s3Request: S3Request, userSTS: User): Future[Done] = {
+    // Start futures concurrently
+    val lineage = createAtlasLineage(response, httpRequest, userSTS, s3Request.clientIPAddress)
+    val notification = createBucketNotification(response, httpRequest, s3Request, userSTS)
+    val permissions = updateBucketPermissions(httpRequest, s3Request)
+
+    // Set handlers to log errors
+    lineage.andThen({
+      case Failure(err) => logger.error(s"Error during lineage creation: ${err}")
+    })
+    notification.andThen({
+      case Failure(err) => logger.error(s"Error while emitting bucket notification: ${err}")
+    })
+    permissions.andThen({
+      case Failure(err) => logger.error(s"Error while setting bucket permissions: ${err}")
+    })
+
+    // Return cumulative future
+    for (
+      _ <- lineage;
+      _ <- notification;
+      _ <- permissions
+    ) yield Done
```

Comment:
because the method always rerturns `Done` it is more readable to return `Unit`

Reply:
`Done` is used to indicate the termination of side effectful computations. I agree that it's the same thing as `Unit` though, just different name. I'll change it.